lib/Runtime/Language/JavascriptExceptionOperators.cpp

```diff
@@ -682,50 +682,46 @@ namespace Js
         // NOTE Don't set the throwing exception here, because we might need to box it and will cause a nested stack walker
         // instead, return it to be set in WalkStackForExceptionContext
 
-        if (stackCrawlLimit == 0)
+        JavascriptExceptionContext::StackTrace *stackTrace = RecyclerNew(scriptContext.GetRecycler(), JavascriptExceptionContext::StackTrace, scriptContext.GetRecycler());
+        if (stackCrawlLimit > 0)
```

Comment:
> You moved RecyclerNew(...StackTrace...) outside of BEGIN/END_TRANSLATE_EXCEPTION block. 

I had checked couple of usages of RecyclerNew and it had looked fine reaching outside of the block. 

I guess the intention is to make this part safe from any possible throw. @jianchun Thanks for the review. I've updated the PR and kept that in.